### ISSUE TITLE
Data race in Tx.Cursor/Executor.Execute

### ISSUE THREAD

**Issue by jwilder** [influxdb/influxdb#3665]
To reproduce:
Enable `TestServer_ContinuousQuery_Deadlock`

`go test -run TestServer_ContinuousQuery_Deadlock -v -race ./cmd/influxd/run/`
 
```
=================
WARNING: DATA RACE
Read by goroutine 49:
  runtime.mapaccess1_faststr()
      /usr/local/go/src/runtime/hashmap_fast.go:179 +0x0
  github.com/influxdb/influxdb/tsdb/engine/b1.(*Tx).Cursor()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/engine/b1/b1.go:539 +0x1cd
  github.com/influxdb/influxdb/tsdb.(*LocalMapper).Open()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:207 +0x1214
  github.com/influxdb/influxdb/tsdb.(*Executor).executeAggregate()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/executor.go:358 +0x630

Previous write by goroutine 47:
  [failed to restore the stack]

Goroutine 49 (running) created at:
  github.com/influxdb/influxdb/tsdb.(*Executor).Execute()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/executor.go:90 +0x147
  github.com/influxdb/influxdb/tsdb.(*QueryExecutor).executeSelectStatement()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/query_executor.go:261 +0xba
  github.com/influxdb/influxdb/tsdb.func·008()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/query_executor.go:146 +0x97a

Goroutine 47 (finished) created at:
  github.com/influxdb/influxdb/cluster.(*PointsWriter).writeToShard()
      /Users/jason/go/src/github.com/influxdb/influxdb/cluster/points_writer.go:271 +0x42c
  github.com/influxdb/influxdb/cluster.func·001()
      /Users/jason/go/src/github.com/influxdb/influxdb/cluster/points_writer.go:205 +0x149
==================
```

**Comment by otoolep** [comment]
Fixed @DanielMorsing ?

**Comment by DanielMorsing** [comment]
closed by #3671
